webfacade/src/index.js

```diff
@@ -57,6 +62,34 @@ export const newConnection = (socket) => {
       socket.emit('register', { succeed: false, message: 'Give both username, email and password' });
     }
   });
+
+  socket.on('subscribeUser', (name) => {
+    client.request('query subscribeUser($name: String!){ subscribeUser (name: $name) { subscribed } }', name)
+      .then((x) => {
+        if (x !== 0) {
+          socket.emit('subscribeUser', { succeed: false, message: 'Cannot subscribe!' });
+        } else {
+          rp({
```

Comment:
Look how you can chain Promises, promises should solve callback hell. When you chain promises you don’t have the same catch statement twice 🌝